src/h5web/dataset-visualizer/VisProvider.tsx

```diff
@@ -5,36 +5,59 @@ import React, {
   useContext,
   useMemo,
 } from 'react';
-import { DimensionMapping } from './models';
+import { transpose } from 'mathjs';
+import { isNumber } from 'lodash-es';
+import { DimensionMapping, Vis } from './models';
 import { HDF5Dataset, HDF5Value } from '../providers/models';
 import { useValue } from '../providers/hooks';
 import { isSimpleShape } from '../providers/utils';
 
 export interface VisProps {
-  mapping: DimensionMapping;
-  rawValues: HDF5Value;
+  visValues: HDF5Value;
   rawDims: number[];
+  slicingIndices?: number[];
 }
 
 export const VisContext = createContext<VisProps | undefined>(undefined);
 
 type Props = {
-  mapping: DimensionMapping;
+  activeVis: Vis;
+  mapperState?: DimensionMapping;
   dataset: HDF5Dataset;
   children: ReactNode;
 };
 
 function VisProvider(props: Props): ReactElement {
-  const { dataset, mapping, children } = props;
+  const { dataset, mapperState, children } = props;
   const rawValues = useValue(dataset.id);
   const rawDims = isSimpleShape(dataset.shape) ? dataset.shape.dims : [];
 
-  if (rawValues === undefined) {
+  const xDim = mapperState ? mapperState.indexOf('x') : -1;
```

Comment:
If there is no mapperState, setting `xDim` to `-1` (or `null`) sets the `mappedValues` to the `rawValues`. It is a bit awkward right now but it will be improved when more Vis/dims are supported.